### PR TITLE
Add analytical derivatives for `LieGroup.log_map()`

### PR DESCRIPTION
**Pull request by fantaosha**
## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have completed my CLA (see **CONTRIBUTING**)
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by fantaosha** [comment]
Now the computation of `SO3.log_map()` and `SE3.log_map()` has been improved around `pi` and all the analytical derivatives tests are passed.

File: theseus/geometry/point_types.py [hunk 1]
```diff
@@ -91,7 +91,7 @@ def exp_map(
         if tangent_vector.ndim != 2 or tangent_vector.shape[1] != 2:
             raise ValueError("Tangent vectors of Point2 should be 2-D vectors.")
 
-        Vector._exp_map_jacobian(tangent_vector, jacobians)
+        Vector._exp_map_jacobian_impl(tangent_vector, jacobians)
```
Comment (mhmukadam) [inline theseus/geometry/point_types.py:94]: Should there be functions in Point2/3 for logmap calling `Vector._log_map_jacobian_impl`?

**Review by mhmukadam — APPROVED** [review]

File: theseus/geometry/point_types.py (hunk 1 above)
Comment (fantaosha) [reply theseus/geometry/point_types.py:94]: Not necessary since `Point2.log_map()` and `Point3.log_map()` calls `Vector.log_map()`

Comment (mhmukadam) [reply theseus/geometry/point_types.py:94]: Ah right. For exp map you needed it explicitly here for the tangent vector size check.

Comment (fantaosha) [reply theseus/geometry/point_types.py:94]: Yes, we checked the sizes [here](https://github.com/facebookresearch/theseus/blob/e487c03917ad19747c2336bdd39f8dbe1d8b70d5/theseus/geometry/point_types.py#L91).

File: theseus/geometry/tests/common.py [hunk 2]
```diff
@@ -259,3 +275,19 @@ def exp_func(xi):
     ]
 
     assert torch.allclose(actual[0], expected, atol=atol)
+
+
+def check_projection_for_vector_log_map(tangent_vector, Group, atol=1e-8):
```
Comment (luisenp) [inline theseus/geometry/tests/common.py:280]: We could reuse `check_projection_for_log_map()` but pass an extra kwarg that specifies if we should call `group.project()` before or not. I think that's the only difference, right?

**Review by luisenp — APPROVED** [review]
LGMT! Left only a very minor comment.

File: theseus/geometry/tests/common.py (hunk 2 above)
Comment (fantaosha) [reply theseus/geometry/tests/common.py]: The `check_projection_for_map` is reused.

**Pull request merged**